File: java/SQL_Injection.java
import java.sql.*;

public class SQL_Injection {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.out.println("Usage: java App <username>");
            return;
        }

        String userInput = args[0];
        String query = "SELECT * FROM users WHERE username='" + userInput + "'";

        try {
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/mydatabase", "myuser", "mypassword");
            Statement stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery(query);

            while (rs.next()) {
                System.out.println(rs.getString("username") + ": " + rs.getString("email"));
            }

            rs.close();
            stmt.close();
            conn.close();
        } catch (SQLException e) {
            System.out.println("SQL error: " + e.getMessage());
        }
    }
}

File: python/SQL_injection.py
import sqlite3

def unsafe_function(user_input):
    connection = sqlite3.connect('example.db')
    cursor = connection.cursor()
    query = f"SELECT * FROM users WHERE username='{user_input}'"
    cursor.execute(query)
    result = cursor.fetchall()
    connection.close()
    return result

user_input = input("Enter a username: ")
print(unsafe_function(user_input))